src/test/java/com/rocketden/main/api/ProblemTests.java

```diff
@@ -39,16 +42,37 @@
     @Autowired
     private MockMvc mockMvc;
 
+    private static final String GET_PROBLEM = "/api/v1/problems/%s";
     private static final String GET_PROBLEM_ALL = "/api/v1/problems";
     private static final String POST_PROBLEM_CREATE = "/api/v1/problems";
+    private static final String POST_TEST_CASE_CREATE = "/api/v1/problems/%s/test-case";
 
     private static final String NAME = "Sort an Array";
     private static final String DESCRIPTION = "Sort an array from lowest to highest value.";
     private static final String NAME_2 = "Find Maximum";
     private static final String DESCRIPTION_2 = "Find the maximum value in an array.";
 
+    private static final String INPUT = "[1, 2, 8]";
+    private static final String OUTPUT = "8";
+    private static final String INPUT_2 = "[-1, 5, 0, 3]";
+    private static final String OUTPUT_2 = "5";
+
     @Test
-    public void createProblemSuccess() throws Exception {
+    public void getProblemNonExistent() throws Exception {
+        ApiError ERROR = ProblemError.NOT_FOUND;
+
+        MvcResult result = this.mockMvc.perform(get(String.format(GET_PROBLEM, 99)))
+                .andDo(print()).andExpect(status().is(ERROR.getStatus().value()))
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        ApiErrorResponse actual = UtilityTestMethods.toObject(jsonResponse, ApiErrorResponse.class);
+
+        assertEquals(ERROR.getResponse(), actual);
+    }
+
+    @Test
+    public void createAndGetProblemSuccess() throws Exception {
```

Comment:
If you allow the user to create a test case with a provided `hidden` value, could you include a test for that as well?

Reply:
Sounds good - there should be a test below and in the service tests that cover that case

Reply:
Ah great, thanks!